### PR TITLE
Docs: Added button and sheet for iOS Android

### PR DESCRIPTION
**Pull request by hectoid**
for button and sheet, both iOS and Android

Thanks for creating a PR! Please follow this template and delete items/sections that are not relevant to your changes, including these instructions.

Please also make sure your [PR title](https://github.com/pinterest/gestalt/#releasing) matches our format: `{ComponentName}-{platform}: Description`

#### What changed?

From a high level, what are the changes this PR introduces? (No need to recount line-by-line, we can see that.)

#### Why?

What is the purpose of this PR? Please include the context around these changes for Future Us. In addition to _what_ is changing, we need to know _why_ these changes are needed. Imagine someone is looking at these changes a year from now and needs to know why they were made.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-XXXX)
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/markdown/Android/sheet.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+## Mobile best practices
+
+- Sheets should be triggered by the users action, like tapping on a button.
+- Use a sheet for non-imersive content and simple tasks.
+- Sheets always sit in front of other UI elements.
+
+<TwoCol>
+  <Group>
+    <ImgContainer src="https://i.pinimg.com/originals/84/d4/0d/84d40d290128608db361a276a586da87.png" alt="a partial sheet with a close button and a grabber to allow for multiple ways to close and collapse"
```
Comment (rlingineni) [inline docs/markdown/Android/sheet.md:36]: @hectoid  - could you check the `ImgContainers` with the do's. They are missing closing tags

**Review by rlingineni — CHANGES_REQUESTED** [review]

File: docs/markdown/Android/sheet.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+By default, Sheet animates in from the bottom of the screen. It animates out when the header close button is pressed, the user swipes down or the user taps outside of the sheet. Visit Material Design for more information on [container motion](https://material.io/design/motion/the-motion-system.html#container-transform).
+
+## Accessibility
+
+People use Android’s accessibility features, such as TalkBack and dynamic text sizing to personalize how they interact with their device. Supporting these personalizations ensures that everyone has a great user experience. See Material Design and development documentation about accessibility for Android:
+
+- [Accessible design on Android](https://material.io/design/usability/accessibility.html#understanding-accessibility)
+- [Accessible development on Android](hhttps://developer.android.com/guide/topics/ui/accessibility)
+
+{/* ## Related
```
Comment (ponori) [inline docs/markdown/Android/sheet.md:117]: Probably best to just remove any comments in these markdown files.

File: docs/markdown/iOS/button.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+- Avoid using multiple button sizes in the same experience.
+- If necessary, adjust the button placement and size when scaling from large screens to small screens.
+
+For general Button best practices, refer to the [Button web documentation](/web/button).
+
+## Variants
+
+### Size
+
+Mobile buttons are available in 2 sizes. The Button text always use [[$font-size-300 token](/foundations/design_tokens#Font-size)] (16pt).
```
Comment (ponori) [inline docs/markdown/iOS/button.md:22]: Remove double bracket.

File: docs/markdown/iOS/sheet.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+By default, Sheet animates in from the bottom of the screen. It animates out when the header close button is pressed, the user swipes down or the user taps outside of the sheet. Visit Apple’s HIG for more information on [Animation and Motion](https://developer.apple.com/design/human-interface-guidelines/foundations/motion/).
+
+## Accessibility
+
+People use Apple’s accessibility features, such as reduced transparency, VoiceOver, and increased text size to personalize how they interact with their device. Supporting these personalizations ensures that everyone has a great user experience. See Apple’s Human Interface Guidelines and API documentation about accessibility for iOS:
+
+- [Accessible design on iOS](https://developer.apple.com/design/human-interface-guidelines/foundations/accessibility/)
+- [Accessible development on iOS](https://developer.apple.com/accessibility/)
+
+{/* ## Related
```
Comment (ponori) [inline docs/markdown/iOS/sheet.md:117]: Remove

File: docs/markdown/iOS/sheet.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
+---
+title: Sheet
```
Comment (ponori) [inline docs/markdown/iOS/sheet.md:2]: It looks like we use a different title for the equivalent doc on Android. Is this intentional?

File: docs/docs-components/siteIndex.js [hunk 6]
```diff
@@ -42,11 +42,11 @@ const siteIndex: $ReadOnlyArray<siteIndexType> = [
   },
   {
     sectionName: 'Android',
-    pages: ['Overview', 'Avatar', 'IconButton', 'Tabs'],
+    pages: ['Overview'],
   },
   {
     sectionName: 'iOS',
-    pages: ['Overview', 'Avatar', 'IconButton', 'Tabs'],
+    pages: ['Overview', 'Avatar', 'IconButton', 'Tabs', 'Button', 'Sheet'],
```
Comment (ponori) [inline docs/docs-components/siteIndex.js:49]: We should probably alphabetize the components.

File: docs/markdown/ios/button.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+For writing best practices, refer to the [web Button documentation](/web/button).
+
+## Accessibility
+
+People use Apple’s accessibility features, such as reduced transparency, VoiceOver, and increased text size to personalize how they interact with their device. Supporting these personalizations ensures that everyone has a great user experience. See Apple’s Human Interface Guidelines and API documentation about accessibility for iOS:
+
+- [Accessible design on iOS](https://developer.apple.com/design/human-interface-guidelines/foundations/accessibility/)
+- [Accessible development on iOS](https://developer.apple.com/accessibility/)
+
+## Internationalization
```
Comment (ponori) [inline docs/markdown/ios/button.md:75]: I'm uncertain if we use Internationalization or Localization within Pinterest. We can loop back to this though if need be.

**Review by ponori — APPROVED** [review]
Nothing blocking, but a few nits to look at if there's time.

**Comment by ponori** [comment]
@hectoid or @rlingineni  One other thing, we'll need to update the docs-components/COMPONENT_DATA.js file to flag Button and Sheet as having documentation so they show up in the Components Overview pages.

**Comment by hectoid** [comment]
@ponori I'll update alphabetization and component_data.js file. You are right: it's "localization" not "internationalization". will update as well.

File: docs/markdown/ios/button.md (hunk 7 above)
Comment (hectoid) [reply docs/markdown/ios/button.md:75]: localization

File: docs/docs-components/siteIndex.js (hunk 6 above)
Comment (hectoid) [reply docs/docs-components/siteIndex.js:49]: will do

**Review by ponori — APPROVED** [review]
Looks great! Let's ship it!

**Review by rlingineni — APPROVED** [review]

**Pull request merged**